File: packages/python/src/cpf_and_cnpj_generator/cpf_generator.py
import random

def random_num():
    # generates a random number between 0 and 9
    return random.randint(0, 9)

def calculate_digit(cpf_list, length, factor):
    _sum = 0

    for i in range(length):
        _sum += cpf_list[i] * (factor - i)
    
    remainder = _sum % 11

    return 0 if remainder < 2 else 11 - remainder

def generate_cpf():
    # Generates a list of 9 random numbers
    cpf_list = [random_num() for _ in range(9)] + [0, 0]

    cpf_list[9] = calculate_digit(cpf_list, 9, 10)

    cpf_list[10] = calculate_digit(cpf_list, 10, 11)

    return ''.join(map(str, cpf_list))

File: packages/typescript/src/cpfGenerator.ts
/**
 * Generates a valid CPF (Brazilian identification number).
 * @returns Returns a valid CPF
 * @description - Made by: Logan
 */
function geraEValida(): string {
  const cpfArray: number[] = new Array<number>(11);

  // Generates an array of 9 random numbers
  for (let i: number = 0; i < 9; i++) {
    cpfArray[i] = (Math.random() * 10) | 0;
  }

  // Calculates the first verifier digit
  let sum: number = 0;
  for (let i: number = 0; i < 9; i++) {
    sum += cpfArray[i] * (10 - i);
  }
  cpfArray[9] = sum % 11 < 2 ? 0 : 11 - (sum % 11);

  // Calculates the second verifier digit
  sum = 0;
  for (let i: number = 0; i < 10; i++) {
    sum += cpfArray[i] * (11 - i);
  }
  cpfArray[10] = sum % 11 < 2 ? 0 : 11 - (sum % 11);

  // Returns the generated CPF as a string
  return cpfArray.join("");
}

export default geraEValida;
